sql/core/src/main/scala/org/apache/spark/sql/GroupedData.scala

```diff
@@ -23,9 +23,28 @@ import scala.language.implicitConversions
 import org.apache.spark.annotation.Experimental
 import org.apache.spark.sql.catalyst.analysis.Star
 import org.apache.spark.sql.catalyst.expressions._
-import org.apache.spark.sql.catalyst.plans.logical.Aggregate
+import org.apache.spark.sql.catalyst.plans.logical.{Rollup, Cube, Aggregate}
 import org.apache.spark.sql.types.NumericType
 
+/**
+ * The Grouping Type
+ */
+sealed private[sql] trait GroupType
```

Comment:
But this will be called from the `DataFrame`, e.g.
```
class DataFrame{
   def groupBy(xx) = new GroupedData(xx).withGroupType(GroupByType)
}
// or
class DataFrame{
   def groupBy(xx) = new GroupedData(xx, GroupByType)
}
```
Either way we have to let the `DataFrame` knows the `GroupType` definition, right?